### DISCUSSION TITLE
Issues with new (WC8) Cart and Checkout block templates

### DISCUSSION THREAD

**Discussion by webmandesign** [woocommerce/woocommerce#44937] [WooCommerce, Blocks, and Site Editing]
Hi,

I'm really confused about introducing the Cart and Checkout templates for block themes. I don't seem to find benefits of these templates.

Can you please provide info why this new approach was introduced? What are the benefits of having these templates in comparison to pages (pre-WC8 approach)?

Reacting to [@ralucaStan feedback](https://github.com/woocommerce/woocommerce-blocks/issues/10595#issuecomment-1683869402), here is my opinion:

I can see these issues with Cart and Checkout templates:

---

### Misused theme template hierarchy?

Templates in WordPress are meant to display multiple occurrences of a similar type of content. For example, the `single.html` template displays single posts and custom post types, `archive.html` template displays archives, `404.html` template displays whenever a URL is not found (error 404), `single-product.html` template displays content of product page.

These follow [WP template hierarchy](https://developer.wordpress.org/themes/basics/template-hierarchy/) very well. All of these mean a _single_ template file displays _multiple_ occurrences of a specific content.

However, the new WC8 Cart and Checkout templates do not seem to follow the same logic - they simply display a content of pages that occur just once. They **behave like pages**, not like templates.

---

### Misused benefits of site editing?

If a user wants to control entire Cart/Checkout page layout including the site header and footer (benefits of site editing), user can simply create a [custom template](https://developer.wordpress.org/themes/advanced-topics/theme-json/#defining-custom-templates) and [use that custom template for their Cart/Checkout page](https://github.com/woocommerce/woocommerce-blocks/issues/10654#issuecomment-1683407598). This is what custom templates are for and there is no need for a theme level Cart/Checkout template here.

---

### Why not to use Cart/Checkout block directly in the page content?

If the Cart and Checkout templates are only to display a specific `woocommerce/cart` and `woocommerce/checkout` blocks, why not to have these blocks inserted directly into the "Cart" and "Checkout" page content? Basically the same approach used with classic themes (or pre-WC8) but instead of shortcodes we can now use blocks:

1. Create a page and name it "Cart".
2. Add [`woocommerce/cart` block into the page content](https://gist.github.com/webmandesign/ad436950ee8596061bbc3696a2b743eb). (Instead of a [classic content with a shortcode](https://gist.github.com/webmandesign/aa21cb7f486cbd76ba3e349c8eb8a7d0).)
3. Publish the page.

Doesn't it work this way? Why a theme level template is required for this approach?

---

### Shouldn't new WC blocks and templates replace the old PHP templates?

If I understand correctly, the new block templates in WooCommerce are meant to replace [old PHP templates](https://github.com/woocommerce/woocommerce/tree/8.0.2/plugins/woocommerce/templates) content with blocks and thus allow users more control without coding. That's perfectly fine and I am very happy about this approach.

But the Cart and Checkout templates seems odd here. I can't find their PHP template equivalent.

---

I've also created some issues regarding Cart and Checkout templates: #10654, #10642 and #10643

**Comment by dkjensen** [comment]
I agree, I find it annoying that WooCommerce templates do not adhere to the default template heirarchy templates I have defined within my theme `templates` directory.

**Comment by Micemade** [comment]
In my theme I have a `page-cart.html` and `page-checkout.html`, but the frontend is displaying default WC templates (WooCommerce 8.0.3 active). Theme cart and checkout templates are listed in Site Editor Templates, as well as the default ones.

When I rename templates to `cart.html` and `checkout.html`, those theme templates do override WC default templates, but I believe we cannot expect that users will be able to rename template names in theme folder.

So, now, my existing theme users have broken Cart and Checkout templates (as it currently is in my theme - the Checkout template is even missing the header template part).

Is there perhaps a way to solve it by making aliases for cart.html / page-cart.html and checkout.html / page-checkout.html?

**Comment by mohsinkhanwebdeveloper** [comment]
hy iu also have same issue how we can edit cart and checkout page with same like old method where we can easily customize with cart.php or formcheckout.php  but now it show wc-block I can make it work in custom code ?
